### PR TITLE
[LUM-860] Upgrade to new JS SDK

### PR DESCRIPTION
**Pull request by ThibaultJRD**
### Introduction

This PR upgrades all the chain bridge to new LUM JS SDK

### Testing

Test everything again

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,7 +34,8 @@
     "dependencies": {
         "@bull-board/api": "^5.9.1",
         "@bull-board/express": "^5.9.1",
-        "@lum-network/sdk-javascript": "^0.8.9",
+        "@lum-network/sdk-javascript": "^1.0.0-beta.2",
+        "@lum-network/sdk-javascript-legacy": "npm:@lum-network/sdk-javascript@0.8.9",
```
Comment (Segfaultd) [inline package.json:38]: Remove legacy

File: src/async/consumers/asset.consumer.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { Logger } from '@nestjs/common';
 import { ConfigService } from '@nestjs/config';
 
 import dayjs from 'dayjs';
-import { ProposalStatus } from '@lum-network/sdk-javascript/build/codec/cosmos/gov/v1beta1/gov';
+import { ProposalStatus } from '@lum-network/sdk-javascript/build/codegen/cosmos/gov/v1/gov';
```
Comment (Segfaultd) [inline src/async/consumers/asset.consumer.ts:6]: We use v1beta1

File: src/async/consumers/block.consumer.ts [hunk 3]
```diff
@@ -3,12 +3,17 @@ import { InjectQueue, Process, Processor } from '@nestjs/bull';
...
 import dayjs from 'dayjs';
-import { LumConstants, LumMessages, LumRegistry, LumUtils } from '@lum-network/sdk-javascript';
+import { LumRegistry, parseRawLogs, sha256, toHex, toJSON, MICRO_LUM_DENOM } from '@lum-network/sdk-javascript';
+import { MsgMultiSend } from '@lum-network/sdk-javascript/build/codegen/cosmos/bank/v1beta1/tx';
 
 import { BlockEntity, TransactionEntity } from '@app/database';
 import { BlockService, ChainService, TransactionService, ValidatorService } from '@app/services';
 import { AssetSymbol, getAddressesRelatedToTransaction, isBeam, NotificationChannels, NotificationEvents, QueueJobs, QueuePriority, Queues } from '@app/utils';
 
+(BigInt.prototype as any).toJSON = function () {
```
Comment (Segfaultd) [inline src/async/consumers/block.consumer.ts:13]: It has nothing to do here

File: src/async/consumers/block.consumer.ts [hunk 4]
```diff
@@ -40,63 +45,62 @@ export class BlockConsumer {
...
             // Fetch and format transactions data
-            const getFormattedTx = async (rawTx: Uint8Array): Promise<Partial<TransactionEntity>> => {
-                // Acquire raw TX
-                const tx = await this._chainService.getChain(AssetSymbol.LUM).client.getTx(LumUtils.sha256(rawTx));
+            const getFormattedTx = async (rawTx: any): Promise<Partial<TransactionEntity>> => {
+                const hash = toHex(sha256(rawTx)).toUpperCase();
 
-                // Decode TX to human-readable format
-                const txData = LumRegistry.decodeTx(tx.tx);
+                console.log(hash);
```
Comment (Segfaultd) [inline src/async/consumers/block.consumer.ts:73]: ?

File: src/async/consumers/block.consumer.ts [hunk 5]
```diff
@@ -40,63 +45,62 @@ export class BlockConsumer {
...
+                // Acquire raw TX
+                const tx = await this._chainService.getChain(AssetSymbol.LUM).client.cosmos.tx.v1beta1.getTx({ hash: hash });
 
                 // Parse the raw logs
-                const logs = LumUtils.parseRawLogs(tx.result.log);
+                const logs = parseRawLogs(tx.txResponse.rawLog);
+
+                console.log('Logs: ', logs);
```
Comment (Segfaultd) [inline src/async/consumers/block.consumer.ts:80]: ?

File: src/async/schedulers/validator.scheduler.ts [hunk 6]
```diff
@@ -2,11 +2,13 @@ import { Injectable, Logger } from '@nestjs/common';
 import { Cron, CronExpression } from '@nestjs/schedule';
 import { ConfigService } from '@nestjs/config';
 
-import { LumConstants, LumRegistry, LumTypes, LumUtils } from '@lum-network/sdk-javascript';
+import { toBech32, toHex, LumBech32Prefixes, fromBech32 } from '@lum-network/sdk-javascript';
+import { LumRegistry as LumLegacyRegistery } from '@lum-network/sdk-javascript-legacy';
```
Comment (Segfaultd) [inline src/async/schedulers/validator.scheduler.ts:6]: Remove legacy

File: src/http/controllers/accounts.controller.ts [hunk 7]
```diff
@@ -133,7 +129,7 @@ export class AccountsController {
                     total: rewards.total.map((rwd) => {
                         return {
                             denom: rwd.denom,
-                            amount: parseInt(rwd.amount, 10) / CLIENT_PRECISION,
+                            amount: parseInt(rwd.amount, 10),
```
Comment (Segfaultd) [inline src/http/controllers/accounts.controller.ts:132]: Where did the precision go?

File: src/http/controllers/accounts.controller.ts [hunk 8]
```diff
@@ -142,13 +138,12 @@ export class AccountsController {
                             reward: rwd.reward.map((rwd2) => {
                                 return {
                                     denom: rwd2.denom,
-                                    amount: parseInt(rwd2.amount, 10) / CLIENT_PRECISION,
+                                    amount: parseInt(rwd2.amount, 10),
```
Comment (Segfaultd) [inline src/http/controllers/accounts.controller.ts:141]: Where did the precision go?

File: src/services/chains/generic.chain.ts [hunk 9]
```diff
@@ -88,27 +102,41 @@ export class GenericChain {
...
+            rpcEndpoint: useEndpoint,
+        });
+
+        this._queryClient = queryClient;
+        this._ibcQueryClient = ibcQueryClient;
+
         // Bind and acquire the chain id
-        this._client = await LumClient.connect(useEndpoint, (error) => {
+        this._chainId = (await queryClient.cosmos.base.tendermint.v1beta1.getNodeInfo()).nodeInfo?.network || 'lum-network-1';
+        const legacyClient = await LumLegacyClient.connect(useEndpoint, (error) => {
```
Comment (Segfaultd) [inline src/services/chains/generic.chain.ts:119]: Remove legacy client

**Review by Segfaultd — CHANGES_REQUESTED** [review]

**Comment by Segfaultd** [comment]
Please make proper PR description.

File: src/http/controllers/accounts.controller.ts (hunk 7 above)
Comment (ThibaultJRD) [reply src/http/controllers/accounts.controller.ts:132]: We don't need this anymore

**Pull request merged**